perun-core/src/main/java/cz/metacentrum/perun/core/impl/HTMLParser.java

```diff
@@ -0,0 +1,162 @@
+package cz.metacentrum.perun.core.impl;
+
+import org.owasp.html.CssSchema;
+import org.owasp.html.HtmlChangeListener;
+import org.owasp.html.HtmlPolicyBuilder;
+import org.owasp.html.PolicyFactory;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+
+/*
+ * Class for sanitizing HTML input using OWASP Java HTML Sanitizer.
+ * @author: Matej Hakoš <492968@mail.muni.cz>
+ */
+public class HTMLParser {
+	private static final String[] allowedTags = {"a", "article", "aside", "b", "blockquote", "br", "button", "caption", "center", "cite", "decorator", "del", "details", "div", "em", "footer", "h1", "h2", "h3", "h4", "h5", "h6", "header", "i", "img", "kbd", "label", "li", "ol", "p", "pre", "section", "select", "span", "strong", "sup", "table", "tbody", "td", "textarea", "tfoot", "th", "thead", "tr", "ul"};
+	private static final String[] allowedAttributes = {"align", "class", "color", "disabled", "height", "hidden", "href", "id", "label", "size", "span", "src", "srcset", "style", "width"};
+	private static final String[] allowedStyles = {"color", "background-color", "font-size", "font-family", "text-align", "margin", "padding", "border", "width", "height", "display", "position", "top", "bottom", "left", "right", "overflow", "float", "clear", "z-index"};
+	private static PolicyFactory policy = null;
+
+	private static List<String> escapedTags = new ArrayList<>();

```

Comment:
i think this might be problematic in case several threads call sanitize method - they all would change the same list and there would be inconsistency in the results. @xflord has a nice idea to make this class builder-like, where every sanitization would create separate class instance. we can discuss it further